python/pyarrow/_dataset.pyx

```diff
@@ -1057,6 +1057,23 @@ cdef class FileFragment(Fragment):
         Fragment.init(self, sp)
         self.file_fragment = <CFileFragment*> sp.get()
 
+    def __repr__(self):
+        type_name = frombytes(self.fragment.type_name())
+        if type_name != "parquet":
+            typ = " type={0}".format(type_name)
+        else:
+            # parquet has a subclass -> type embedded in class name
+            typ = ""
+        partition_dict = _get_partition_keys(self.partition_expression)
+        partition = ", ".join(
+            ["{0}={1}".format(key, val) for key, val in partition_dict.items()]
```

Comment:
nit: this might be a little clearer with f-strings over str.format

Reply:
I always forget in which Python version that was introduced, but it's Python 3.6, so indeed we can use f-strings!